spec/puppet/resource_api/simple_provider_spec.rb

```diff
@@ -177,4 +183,25 @@ def delete(context, _name); end
       provider.set(context, changes)
     end
   end
+
+  context 'with a type that does not implement ensurable' do
+    let(:is_values) { { name: 'title', content: 'foo' } }
+    let(:should_values) { { name: 'title', content: 'bar' } }
+    let(:changes) do
+      { 'title' =>
+            {
+              is: is_values,
+              should: should_values,
+            } }
+    end
+
+    before(:each) do
+      allow(context).to receive(:updating).with('title').and_yield
+      allow(context).to receive(:feature_support?).with('simple_get_filter')
```

Comment:
Seeing that here reminds me that `feature_support?` could become `type.has_feature?` (c.f. `hash.has_key?`). Note that since we're already 1.0, we can't just delete `feature_support?` either way.